src/containers/MessageView/MessageView.tsx

```diff
@@ -65,31 +66,35 @@ class MessageView extends React.Component<Props, State> {
         this.fetchMessages()
       })
     }
-
-    safeDelete(
-      <span>
-        the message &apos;
-        <MessageSubject message={message} />
-        &apos; from{' '}
-        <ContactPresentation address={message.senderAddress} inline />
-      </span>,
-      (notification: IBlockingNotification) => {
-        MessageRepository.deleteByMessageId(message.messageId as string)
-          .then(() => {
-            this.fetchMessages()
-            notification.remove()
-          })
-          .catch(error => {
-            errorService.log({
-              error,
-              message: `Could not delete message ${message.messageId}`,
-              origin: 'MessageView.onDeleteMessage()',
-              type: 'ERROR.FETCH.DELETE',
+    if (selectedIdentity) {
+      safeDelete(
+        <span>
+          the message &apos;
+          <MessageSubject message={message} />
+          &apos; from{' '}
+          <ContactPresentation address={message.senderAddress} inline />
+        </span>,
+        (notification: IBlockingNotification) => {
+          MessageRepository.deleteByMessageId(
+            message.messageId as string,
+            selectedIdentity.identity.signStr(message.messageId as string)
```

Comment:
If `messageId` is a number, you can apply [toString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toString) instead of using as

Reply:
messageId is `string | undefined`  
we previously returned on `!messageId`, now we check for existence and do a non-null assertion